### ISSUE TITLE
 AdguardHome ignores `bind_host` and `bind_port` settings.

### ISSUE THREAD

**Issue by fastfailure** [NixOS/nixpkgs#281995]
### Describe the bug

AdguardHome ignores `bind_host` and `bind_port` settings, deprecated in favour of `http.address`.

### Steps To Reproduce
 
Configure adguardhome with `services.adguardhome.settings.bind_port = 3001;` (the default is 3000) and/or `services.adguardhome.settings.bind_host = "127.0.0.1";` (the default is *).

Adguard service listen on on *:3000 instead of 127.0.0.1:3001.

Workaround: use instead `settings.http.address = "127.0.0.1:3001";`

### Additional context

As stated in the [documentation](https://github.com/AdguardTeam/AdGuardHome/wiki/Configuration#configuration-file) `bind_host` and `bind_port` are deprecated after v0.107.33:

> bind_host (before v0.107.33): Web interface IP address to listen on.
> bind_port (before v0.107.33): Web interface IP port to listen on.
> http (since v0.107.36): Web interface configuration.
> -   address: Web interface IP address with port to listen on.

### Notify maintainers
@numkem @iagoq @rhoriguchi
@aanderse @CRTified

### Metadata

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.1.71, NixOS, 23.11 (Tapir), 23.11.2936.428544ae95ee`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.1`
 - channels(root): `"nixos-23.11"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

**Comment by CRTified** [comment]
Should be fixed by #247828

**Comment by eclairevoyant** [comment]
PR merged. please reopen if still an issue.
